Content.Server/StationEvents/Events/StationEventSystem.cs

```diff
@@ -53,6 +54,22 @@ protected override void Added(EntityUid uid, T component, GameRuleComponent game
         if (!TryComp<StationEventComponent>(uid, out var stationEvent))
             return;
 
+        if (stationEvent.BlacklistRules != null && stationEvent.BlacklistRules.Count() >0 )
+        {
+            foreach (var blockedRule in stationEvent.BlacklistRules)
+            {
+                if (EntityManager.ComponentFactory.TryGetRegistration(blockedRule, out var reg))
+                {
+                    if (EntityManager.GetAllComponents(reg.Type).Any())
+                    {
+                        AdminLogManager.Add(LogType.EventStopped, $"The {ToPrettyString(uid)} is not started because there is already a {blockedRule} in the game.");
+                        ForceEndSelf(uid, gameRule);
+                        return;
+                    }
+                }
+            }
+        }
+
```

Comment:
This looks kinda hacky?

Reply:
I don't like the whole structure of events. It feels like they can't be interrupted conveniently under certain conditions. As it happens in a lot of other systems, with args.Canceled

If you have any ideas on how to do blacklisting at events here more correctly, I'm happy to listen

Reply:
Just add a group prototype + count instead.